core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java

```diff
@@ -68,21 +70,29 @@
 
     /**
      * Defines the user language used for building the native executable.
+     * It also serves as te default Locale language for the native executable application runtime.
+     * e.g. en or cs as defined by IETF BCP 47 language tags.
      * <p>
-     * Defaults to the system one.
+     * 
+     * @deprecated Use the global quarkus.default-locale.
      */
-    @ConfigItem(defaultValue = "${user.language:}")
-    @ConvertWith(TrimmedStringConverter.class)
-    public Optional<String> userLanguage;
+    @ConfigItem
+    @ConvertWith(LocaleConverter.class)
+    @Deprecated
+    public Optional<Locale> userLanguage;
```

Comment:
I think you should keep this thing a String as it's just the language part. Also see in another comment how it gets confusing.